#!/usr/bin/env python
PACKAGE = "lidar_tracking"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("file_name", str_t,    0, "file_name to save, change name will save to file",    "crop_out.ply")

gen.add("crop_box_min_x", double_t,    0, "",    -2.0,-50.0,    50)
gen.add("crop_box_max_x", double_t,    0, "",    2.0, -50.0,   50)
gen.add("crop_box_min_y", double_t,    0, "",    -2.0,-10.0,    10)
gen.add("crop_box_max_y", double_t,    0, "",    2.0, -10.0,   10)
gen.add("crop_box_min_z", double_t,    0, "",    -2.0, -10.0,   10)
gen.add("crop_box_max_z", double_t,    0, "",    2.0, -10.0,   10)

gen.add("target_pose_x", double_t,    0, "",    -10.0, -50.0,   50)
gen.add("target_pose_y", double_t,    0, "",    0.0, -10.0,   10)
gen.add("target_pose_z", double_t,    0, "",    1.8, -10.0,   10)
gen.add("target_pose_roll", double_t,    0, "",    0.0, -3.14159,   3.14159)
gen.add("target_pose_pitch", double_t,    0, "",    0.0, -3.14159,   3.14159)
gen.add("target_pose_yaw", double_t,    0, "",    0.0, -3.14159,   3.14159)

# exit(gen.generate(PACKAGE, "node_name", "file_name_prefix"))	#this line will generate a file named prefixConfig.h
exit(gen.generate(PACKAGE, "pcl_editor_node", "PclEditor"))
